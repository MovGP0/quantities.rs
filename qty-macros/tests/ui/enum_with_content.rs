// ---------------------------------------------------------------------------
// Copyright:   (c) 2021 ff. Michael Amrhein (michael@adrhinum.de)
// License:     This program is part of a larger application. For license
//              details please read the file LICENSE.TXT provided together
//              with the application.
// ---------------------------------------------------------------------------
// $Source$
// $Revision$

use qty_macros::VariantsAsConstants;

#[derive(VariantsAsConstants, Debug, PartialEq)]
pub enum TestEnum {
    A(i32),
    B,
}

fn main() {}
