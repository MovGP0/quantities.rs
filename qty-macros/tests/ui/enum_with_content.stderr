error: proc-macro derive panicked
  --> tests/ui/enum_with_content.rs:12:10
   |
12 | #[derive(VariantsAsConstants, Debug, PartialEq)]
   |          ^^^^^^^^^^^^^^^^^^^
   |
   = help: message: The given enum must be a fieldless enum.
